Documentation (#610)
nf-core/sarek: docs/usage.md

The typical command for running the pipeline is as follows:

```console
nextflow run nf-core/sarek --input samplesheet.csv  --outdir <OUTDIR> -profile docker
```

> This will launch the pipeline with the `docker` configuration profile. See below for more information about profiles.

Reviewer: Just wanted to mention it's producing a `.nextflow.log` not `.nextflow_log` below

Reviewer: And i think it should be **Sample sheet**

| `status`  | **Normal/tumor status of sample**; can be `0` (normal) or `1` (tumor).<br /> _Optional, Default: `0`_                                                                                                                                                                                                                             |
| `sample`  | **Custom sample ID** for each tumor and normal sample; more than one tumor sample for each subject is possible, i.e. a tumor and a relapse; samples can have multiple lanes for which the _same_ ID must be used to merge them later (see also `lane`). Sample IDs must be unique for unique biological samples <br /> _Required_ |
| `lane`    | Lane ID, used when the `sample` is multiplexed on several lanes. Must be unique for each lane in the same sample (but does not need to be the original lane name), and must contain at least one character <br /> _Required for `--step_mapping`_                                                                                 |

> | `fastq_1` | Full path to FastQ file for Illumina short reads 1. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".                                                                                                                                                                                                        |

Reviewer: ```suggestion
| `fastq_1` | Full path to FastQ file for Illumina short reads 1. File has to be gzipped and have the extension `.fastq.gz` or `.fq.gz`.                                                                                                                                                                                                        |
```

--input '[path to samplesheet file(s)]'
```

#### Overview: Samplesheet Columns

| Column    | Description                                                                                                                                                                                                                                                                                                                       |
| --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `patient` | **Custom patient ID**; designates the patient/subject; must be unique for each patient, but one patient can have multiple samples (e.g. normal and tumor). <br /> _Required_                                                                                                                                                      |

> | `gender`  | **Sex chromosomes of the patient**; i.e. XX, XY..., only used for Copy-Number Variation analysis in a tumor/pair<br /> _Optional, Default: `NA`_                                                                                                                                                                                  |

Reviewer 2: `gender` is now `sex`

patient1,test_sample,lane_1,test.bam
```

> In this example, the sample is multiplexed over 3 lanes:

Reviewer 3: ```suggestion
In this example, the sample is multiplexed over three lanes:
```

##### Full samplesheet

> In this example, all possible columns are used. There are 3 lanes for the normal sample, 2 for the tumor sample, 1 for the relapse, including the `sex` and `status` information per patient:

Reviewer 3: ```suggestion
In this example, all possible columns are used. There are three lanes for the normal sample, two for the tumor sample, and one for the relapse sample, including the `sex` and `status` information per patient:
```

If you have any questions or issues please send us a message on [Slack](https://nf-co.re/join/slack) on the [`#configs` channel](https://nfcore.slack.com/channels/configs).

## Troubleshooting & FAQ

### How to test the pipeline

### Variantcalling tools

#### Which tool can be used for for which data type?

This list is by no means exhaustive and it will depend on the specific analysis you would like to run. This is a suggestion based on the individual docs of the tools specifically for human genomes and a garden-variety sequencing run.

| Tool            | WGS | WES |  Panel | Tumor |  Normal | Somatic |

> | :-------------- | :-: | :-: | :----: | :---: | :-----: | :-----: |

Reviewer: Maybe we should consider converting all tools to lowercase and change mutect to mutect2

Re-start your session.

Note that the way to increase the open file limit in your system may be slightly different or require additional steps.

#### Cannot delete work folder when using docker + Spark

Currently, when running spark-based tools in combination with docker, it is required to set `docker.userEmulation = false`. This can unfortunately causes permission issues when `work/` is being written with root permissions. In case this happens, you might need to configure docker to run without `userEmulation` (see [here](https://github.com/Midnighter/nf-core-adr/blob/main/docs/adr/0008-refrain-from-using-docker-useremulation-in-nextflow.md)).

> 

Reviewer 4: ```suggestion

### How to handle UMIs

Sarek can process UMI-reads, using [fgbio](http://fulcrumgenomics.github.io/fgbio/tools/latest/) tools.

In order to use reads containing UMI tags as your initial input, you need to include `--umi_read_structure [structure]` in your parameters. 

This will enable pre-processing of the reads and UMI consensus reads calling, which will then be used to continue Sarek workflow.

#### UMI Read Structure

This parameter is a string, which follows a [convention](https://github.com/fulcrumgenomics/fgbio/wiki/Read-Structures) to describe the structure of the umi.
If your reads contain a UMI only on one end, the string should only represent one structure (i.e. "2M11S+T"); should your reads contain a UMI on both ends, the string will contain two structures separated by a blank space (i.e. "2M11S+T 2M11S+T").

#### Limitations and future updates

Recent updates to Samtools have been introduced, which can speed-up performance of fgbio tools used in this workflow.
The current workflow does not handle duplex UMIs (i.e. where opposite strands of a duplex molecule have been tagged with a different UMI), and best practices have been proposed to process this type of data.
Both changes will be implemented in a future release.

```

Re-start your session.

Note that the way to increase the open file limit in your system may be slightly different or require additional steps.

#### Cannot delete work folder when using docker + Spark

Currently, when running spark-based tools in combination with docker, it is required to set `docker.userEmulation = false`. This can unfortunately causes permission issues when `work/` is being written with root permissions. In case this happens, you might need to configure docker to run without `userEmulation` (see [here](https://github.com/Midnighter/nf-core-adr/blob/main/docs/adr/0008-refrain-from-using-docker-useremulation-in-nextflow.md)).

### How to handle UMIs

Sarek can process UMI-reads, using [fgbio](http://fulcrumgenomics.github.io/fgbio/tools/latest/) tools.

In order to use reads containing UMI tags as your initial input, you need to include `--umi_read_structure [structure]` in your parameters.

> This will enable pre-processing of the reads and UMI consensus reads calling, which will then be used to continue Sarek workflow. It is then recommended to skip an additional duplicate marking and base quality score recalibration. See [`--skip_tools`].

Author: @lescai added the last part about skipping of bqsr and md based on our slack messages. Have no other sources for it though. but this is correct, right?

Reviewer 4: argh yes I forgot that :) good catch! this is actually very important, thanks for adding it!

Author: added a vague statement now to at least have users think about it
